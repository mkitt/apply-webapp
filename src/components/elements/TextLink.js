// @flow
import glamorous from 'glamorous'
import Link from './Link'
import { COLORS } from '../../constants/theme'

export default glamorous(Link)(
  {
    position: 'relative',
    zIndex: 2,
    color: COLORS.BLUE,
    textDecoration: 'none',
    transition: 'color 200ms',
  },
  ({ fontSize }) => ({
    fontSize: fontSize || null,
  }),
)
