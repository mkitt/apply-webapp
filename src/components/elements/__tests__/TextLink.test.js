import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import TextLink from '../TextLink'

describe('<TextLink />', () => {
  it('shows the correct snapshot tree for the <TextLink> element', () => {
    const component = (
      <MemoryRouter>
        <TextLink href="https://www.turing.io">The TextLink</TextLink>
      </MemoryRouter>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <TextLink> element with options', () => {
    const component = (
      <MemoryRouter>
        <TextLink to="/hello" color="magenta" fontSize={18} mt={8}>
          The TextLink with options
        </TextLink>
      </MemoryRouter>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
