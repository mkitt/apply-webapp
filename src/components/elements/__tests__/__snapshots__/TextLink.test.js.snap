// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextLink /> shows the correct snapshot tree for the <TextLink> element 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  z-index: 2;
  color: #35c5fc;
  text-decoration: none;
  transition: color 200ms;
  -webkit-transition: color 200ms;
  -moz-transition: color 200ms;
}

<a
  className="css-0"
  href="https://www.turing.io"
  rel="noopener noreferrer"
  target="_blank"
>
  The TextLink
</a>
`;

exports[`<TextLink /> shows the correct snapshot tree for the <TextLink> element with options 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  z-index: 2;
  color: magenta;
  text-decoration: none;
  transition: color 200ms;
  margin-top: 512px;
  font-size: 18px;
  -webkit-transition: color 200ms;
  -moz-transition: color 200ms;
}

<a
  className="css-0"
  color="magenta"
  fontSize={18}
  href="/hello"
  mt={8}
  onClick={[Function]}
>
  The TextLink with options
</a>
`;
